spec/jobs/reports/quarterly_account_stats_spec.rb

```diff
@@ -0,0 +1,81 @@
+require 'rails_helper'
+require 'csv'
+
+RSpec.describe Reports::QuarterlyAccountStats do
+  subject(:report) { described_class.new }
+
+  describe '#perform' do
+    let(:end_date) { Time.zone.today }
+    let(:ninety_days_ago) { end_date - 90.days }
+
+    it 'saves the report' do
+      expect(subject).to receive(:report_body).
+        with(ninety_days_ago, end_date).
+        and_return('csv text')
+      expect(subject).to receive(:save_report).with(
+        'quarterly-account-stats',
+        'csv text',
+        extension: 'csv',
+      )
+      report.perform(Time.zone.today)
+    end
+  end
+
+  describe '#report_body' do
+    subject(:report_body) { report.report_body(start_date, end_date) }
+
+    let(:start_date) { Time.zone.today - 90.days }
+    let(:end_date) { Time.zone.today - 1.day }
+
+    before do
+      # These are older than 90 days and should go to grand totals,
+      # but not counts for the period.
+      travel_to(Time.zone.today - 6.months) do
+        create_accounts
+      end
+
+      # These are recent and should be included in both the period and
+      # grand totals.
+      travel_to(Time.zone.today - 1.week) do
+        create_accounts
+      end
+
+      # This is too new for our interval, so it should only show up in
+      # the grand total, and even that's debatable.
```

Comment:
Trying to write a good test accidentally led to me debating whether I should complicate things further or not. 😇 

This made me realize that, if given a date in the past, the results may be misleading, because the `..._all_time` numbers are counts from the present, regardless of end date. E.g., if you gave a date of January 1, 1990, the `...for_period` counts would show the number of users signing up between the fall of 1989 and the start of 1990 (i.e., 0), but the `..._all_time` number would be the number of accounts _today_, not in 1990.

I am inclined to say we don't support that, and make `perform` accept no parameters, and just always run from the date it's called. My reasoning is:

1. The request was for account creation/deletion/proofing _rate_ over 90 days, and we just decided to include grand totals in case it was useful.
2. I think giving a truly accurate count might be more complicated than it seems. On the surface we could just change the `created_at` parameter in the query, but I think deleted users complicate things, and I'm not convinced it's worth the effort to get right given #1.

If the need emerges, we can always add it in the future.

Reply:
I had actually started down that path, but calling methods to set up deleted users, etc. started to make it feel like I was likely going to miss a timestamp. Wrapping it all in `travel_to` and letting it handle it was a nice safety blanket. This is another one of those, "I don't feel strongly, but it's not obvious to me how that would really help" situations.

That said, this doesn't answer the bigger question here of what the _code_ should do. Do you agree with my suggestion above to not accept past dates rather than complicate the code for a feature that wasn't requested?

Reply:
I think it's very important to accept past dates so we can backfill for previous quarters. I think any report like this we'll get asked to see over time